learning_resources_search/api.py

```diff
@@ -887,11 +891,81 @@ def get_similar_resources(
             minimum times a term needs to show up in input
         min_doc_freq (int):
             minimum times a term needs to show up in docs
+        use_embeddings (bool):
+            use vector embeddings to retrieve results
 
     Returns:
         list of str:
             list of topic values
     """
+    if use_embeddings:
+        return get_similar_resources_qdrant(value_doc, num_resources)
+    return get_similar_resources_opensearch(
+        value_doc, num_resources, min_term_freq, min_doc_freq
+    )
+
+
+def _qdrant_similar_results(doc, num_resources):
+    from learning_resources_search.indexing_api import qdrant_client
+
+    client = qdrant_client()
+    return [
+        hit.metadata
+        for hit in client.query(
+            collection_name=f"{settings.QDRANT_BASE_COLLECTION_NAME}.resources",
+            query_text=(
+                f'{doc.get("title")} {doc.get("description")} '
+                f'{doc.get("full_description")} {doc.get("content")}'
+            ),
+            limit=num_resources,
+        )
+    ]
+
+
+def get_similar_resources_qdrant(value_doc: dict, num_resources: int):
+    """
+    Get a list of similar resources from qdrant
+
+    Args:
+        value_doc (dict):
+            a document representing the data fields we want to search with
+        num_topics (int):
+            number of resources to return
+
+    Returns:
+        list of str:
+            list of learning resources
+    """
+    hits = _qdrant_similar_results(value_doc, num_resources)
+    return LearningResource.objects.for_search_serialization().filter(
```

Comment:
This is resulting in numerous n+1 queries. Adjusting the `for_search_serialization` function to this seems to get rid of them:

```python
    def for_search_serialization(self):
        return self.for_serialization().annotate(in_featured_lists=Count("parents__parent__channel"))
```